changelog.d/2712.change.rst

```diff
@@ -0,0 +1,2 @@
+Added `glob:` directive for use inside `[options.data_files]` values, with accompanying unittests.
+Updated declarative_config.rst docs to explain new `glob:` directive.
```

Comment:
A changelog entry can probably skip mentioning that:
- the change comes with unittests
- the change includes documentation updates

Maybe this info would fit better in the commit message?